blog/2024-11-11-intro-to-idpbuilder/index.md

```diff
@@ -0,0 +1,82 @@
+
+```bash
+idpbuilder create --use-path-routing \
+  --package https://github.com/cnoe-io/stacks//ref-implementation
+```
+
+![img](images/idpbuilder-basic.png)
+
+
+In addition to its simplicity, idpBuilder offers the following key benefits:
+
+1. Rapid Deployment: Create a reference implementation of an IDP with minimal setup time.
+2. CI Integration: Easily incorporate idpBuilder into your continuous integration workflows for comprehensive testing.
+3. Local Development: Provide IDP engineers with a consistent and easily reproducible local development environment.
+
+Let's explore how you can use idpBuilder in your development workflow. Whether you're looking to demonstrate an IDP reference implementation, enhance your CI pipeline, or improve your local development setup, idpBuilder offers a solution that's both powerful and easy to use.
+
+
+## Demo your own IDP implementation
+
+idpBuilder comes with a set of technologies that enables GitOps workflows all contained within the ephemeral environment. Think of building your IDP solution in a box. It does this by provisioning a kind cluster, Gitea server, ArgoCD, and ingress-nginx. See[ our documentation](https://cnoe.io/docs/reference-implementation/installations/idpbuilder/usage) site for more information.
+
+in addition, idpbuilder can copy files and Kubernetes manifests checked into Git repositories to the in-cluster Gitrea repositories. Once copied to the in-cluster repositories, ArgoCD can use them to deploy your solutions in minutes. We have examples of this in the [Stacks repository.](https://github.com/cnoe-io/stacks/tree/main/ref-implementation).  You can run them in your browser using Codespaces as well.
+This approach also allows you to experiment with configuration changes and code changes without changing files checked into the external repositories because everything is contained in the cluster.
+
+Carlos Santana does an excellent job of show this in his video. https://www.youtube.com/watch?v=e6Fvivx4Aw8
+
+
+## Local Development and CI integrations
+
+Many organizations struggle with fragmented development environments. Different teams often work in silos, using disparate tools and workflows. This fragmentation can lead to inefficiencies, slower development cycles, and integration headaches.  idpBuilder can set up identical environments for local development and CI pipelines. With idpBuilder, you get:
+
+1. **Reduced "It Works on My Machine" Syndrome**: With consistent environments, discrepancies between local and CI setups become a thing of the past.
+2. **Early Issue Detection**: Integration problems are caught earlier, saving time and resources in the long run.
+3. **Improved Collaboration**: Developers can confidently work on the same project, knowing they're all using identical environments.
+4. **Streamlined Workflow**: The seamless transition from local development to CI pipelines accelerates the development cycle.
+
```

Comment:
What about adding the something along the lines of being able to work with gitops without creating public git repos, or configuring gitops tools with credentials. Is a local git server ready for gitops

Reply:
@csantanapr Added a bullet point about that